Tests/SphericalExtractionUniformTest/SphericalExtractionUniformTest.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ *      _______              __
+ *     / ___/ /  ___  __ _  / /  ___
+ *    / /__/ _ \/ _ \/  V \/ _ \/ _ \
+ *    \___/_//_/\___/_/_/_/_.__/\___/
+ *    Please refer to LICENSE, in Chombo's root directory.
+ */
+#endif
+
+// Chombo includes
+#include "parstream.H" //Gives us pout()
+
+// General includes:
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <sys/time.h>
+
+using std::endl;
+#include "DefaultLevelFactory.hpp"
+#include "GRAMR.hpp"
+
+#include "GRParmParse.hpp"
+#include "SetupFunctions.hpp"
+#include "SimulationParameters.hpp"
+
+// Problem specific includes:
+#include "AMRInterpolator.hpp"
+#include "Lagrange.hpp"
+#include "SphericalExtraction.hpp"
+#include "SphericalExtractionUniformTestLevel.hpp"
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+// Chombo namespace
+#include "UsingNamespace.H"
+
+void surface_integration(
+    const spherical_extraction_params_t &extraction_params_lo,
+    const SimulationParameters &sim_params,
+    AMRInterpolator<Lagrange<4>> &interpolator,
+    std::pair<std::vector<double>, std::vector<double>> &integral_lo,
+    std::pair<std::vector<double>, std::vector<double>> &integral_hi,
+    const IntegrationMethod &method)
+{
+    // low resolution spherical extraction
+    SphericalExtractionUniform spherical_extraction_lo(
+        extraction_params_lo, sim_params.coarsest_dx * sim_params.dt_multiplier,
+        0.0, true, 0.0);
+    spherical_extraction_lo.add_var(c_phi_Re);
+    spherical_extraction_lo.add_var(c_phi_Im);
+    spherical_extraction_lo.extract(&interpolator);
+    spherical_extraction_lo.write_extraction("ExtractionOutLo_");
+
+    // high resolution spherical extraction
+    spherical_extraction_params_t extraction_params_hi = extraction_params_lo;
+    // we are only checking the converence in theta integration
+    // extraction_params_hi.num_points_phi *= 2;
+    extraction_params_hi.num_points_theta *= 2;
+    SphericalExtractionUniform spherical_extraction_hi(
+        extraction_params_hi, sim_params.coarsest_dx * sim_params.dt_multiplier,
+        0.0, true, 0.0);
+    spherical_extraction_hi.add_var(c_phi_Re);
+    spherical_extraction_hi.add_var(c_phi_Im);
+    spherical_extraction_hi.extract(&interpolator);
+    spherical_extraction_hi.write_extraction("ExtractionOutHi_");
+
+    // real part is the zeroth componenent and imaginary part is first component
+    auto extracted_harmonic = [](std::vector<double> &data, double, double,
+                                 double) {
+        return std::make_pair(data[0], data[1]);
+    };
+
+    // add the spherical harmonic mode integrands for each resolution
+    // Always use trapezium rule in phi as this is periodic
+    bool broadcast_integral = true;
+    spherical_extraction_lo.add_mode_integrand(
+        sim_params.es, sim_params.el, sim_params.em, extracted_harmonic,
+        integral_lo, method, IntegrationMethod::trapezium, broadcast_integral);
```

Comment:
Oh you use the trapezium rule here for the integration in phi but change it to Simpson's rule for the normal SphericalExtractionTest. Is there a reason for this?